library/src/main/java/com/mux/video/upload/internal/UploadMetrics.kt

```diff
@@ -13,14 +13,191 @@ import okhttp3.Interceptor
+  }
+
+  @JvmSynthetic
+  internal suspend fun reportStandardizationSuccess(
+    startTimeMillis: Long,
+    endTimeMillis: Long,
+    inputFileDurationMs: Long,
+    inputReasons: JSONArray,
+    maximumResolution:String,
+    sessionId: String,
+    uploadInfo: UploadInfo
+  ) {
+    var body = JSONObject()
+    body.put("type", "upload_input_standardization_succeeded")
+    body.put("session_id", sessionId)
+    body.put("version", "1")
+    val data = getEventInfo(startTimeMillis, endTimeMillis, inputFileDurationMs, uploadInfo)
+    data.put("maximum_resolution", maximumResolution)
+    data.put("non_standard_input_reasons", inputReasons)
+    body.put("data", data)
+    sendPost(body)
+  }
+
+  @JvmSynthetic
+  internal suspend fun reportStandardizationFailed(
+    startTimeMillis: Long,
+    endTimeMillis: Long,
+    inputFileDurationMs: Long,
+    errorDescription: String,
+    inputReasons: JSONArray,
+    maximumResolution:String,
+    sessionId: String,
+    uploadInfo: UploadInfo
+  ) {
+    var body = JSONObject()
+    body.put("type", "upload_input_standardization_failed")
+    body.put("session_id", sessionId)
+    body.put("version", "1")
+    val data = getEventInfo(startTimeMillis, endTimeMillis, inputFileDurationMs, uploadInfo)
+    data.put("error_description", errorDescription)
+    data.put("maximum_resolution", maximumResolution)
+    data.put("non_standard_input_reasons", inputReasons)
+    data.put("upload_canceled", false)
+    body.put("data", data)
+    sendPost(body)
+  }
+
+  @JvmSynthetic
+  internal suspend fun reportUploadSucceeded(
+    startTimeMillis: Long,
+    endTimeMillis: Long,
+    inputFileDurationMs: Long,
+    sessionId: String,
+    uploadInfo: UploadInfo
+  ) {
+    var body = JSONObject()
+    body.put("type", "upload_succeeded")
+    body.put("session_id", sessionId)
+    body.put("version", "1")
+    val data = getEventInfo(startTimeMillis, endTimeMillis, inputFileDurationMs, uploadInfo)
+    data.put("input_standardization_requested", uploadInfo.standardizationRequested
+            && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
+    body.put("data", data)
+    sendPost(body)
+  }
+
+  @JvmSynthetic
+  internal suspend fun reportUploadFailed(
+    startTimeMillis: Long,
+    endTimeMillis: Long,
+    inputFileDurationMs: Long,
+    errorDescription:String,
+    sessionId: String,
+    uploadInfo: UploadInfo
+  ) {
+    var body = JSONObject()
+    body.put("type", "uploadfailed")
+    body.put("session_id", sessionId)
+    body.put("version", "1")
+    val data = getEventInfo(startTimeMillis, endTimeMillis, inputFileDurationMs, uploadInfo)
+    data.put("input_standardization_requested", uploadInfo.standardizationRequested
+            && Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
+    data.put("error_description", errorDescription)
+    body.put("data", data)
+    sendPost(body)
+  }
+
   @JvmSynthetic
   internal suspend fun reportUpload(
```

Comment:
question(blocking): pretty sure we want to remove this and its invocation/usage in `UploadJobFactory` (including the `BuildType` gate), as it should be replaced by the more verbose, specific `report*` methods you've implemented. cc @daytime-em for confirmation. Related: It looks like we're no longer respecting the `optOut` option for telemetry tracking. Make sure you re-apply that conditional check around any method invocations in `UploadJobFactory`.